torchvision/prototype/models/googlenet.py

```diff
@@ -37,8 +37,10 @@ def googlenet(weights: Optional[GoogLeNetWeights] = None, progress: bool = True,
     original_aux_logits = kwargs.get("aux_logits", False)
     if weights is not None:
         if "transform_input" not in kwargs:
+            # Similar to the comment about num_classes, I think we should raise and error if the user manually specified transform_input=False
```

Comment:
This is actually part of the original implementation, though I tried to simplify its unnecessarily complex logic a bit:
https://github.com/pytorch/vision/blob/72054ca05870174b18a1421479e76a3da5e39782/torchvision/models/googlenet.py#L293-L303

So parameter overwriting is something that used to happen in the past and changing the behaviour now to throw errors will be BC-breaking. cc @fmassa

Reply:
I wouldn't consider this a BC-breaking change, since we're preventing the user from doing something they shouldn't do. I think it's fair to consider this a bug-fix :)

Reply:
So you are saying that if we change our policy and start throwing exceptions, instead of silently overwriting variables, you would be OK and consider it not BC-breaking but a bug fix?

Reply:
yes :)

Reply:
The parameter overwriting is legacy due to the fact of how the model was implemented / ported.
If I could start from scratch I would have just removed `transform_input` altogether and changed the pre-trained weights.

Reply:
@fmassa Understood, we had to do what we had to do.

The question here is whether we will adopt everywhere the policy of throwing exceptions instead of silently overwriting  input parameters with weight specific config. If you recall you brought up this topic in an earlier PR, suggesting that instead of overwriting the `num_classes` we should raise an error if the numbers don't match with the pretrained config. So the question here is, should we do the same for other parameters such as `transform_input`, `aux_logits`, `init_weights` etc.

@NicolasHug proposed to treat this as a bug and thus fix it. Hence we wouldn't consider this BC-breaking. It wouldn't be the first time to do this, so the idea is not crazy.

I'm a bit concerned that this might have downstream effects to users. An alternative approach could be to throw warnings instead of errors. Thoughts?

Reply:
I think silent overrides can be bad and we should avoid it.
```python
resnet50(pretrained=True, num_classes=10)
```
should raise a loud error (which is currently the case in the old code-path).

Reply:
Addressed at PR #4948 commit 9ff811ba6ae4af939aa8fb6f44b0e7b5eaef818a